src/query/ee/src/storages/fuse/operations/vacuum_table_v2.rs

```diff
@@ -0,0 +1,320 @@
+    let gc_root_segments = gc_root
+        .segments
+        .iter()
+        .map(|(path, _)| path)
+        .collect::<HashSet<_>>();
+    let segments_before_gc_root = list_until_timestamp(
+        fuse_table,
+        &fuse_table.meta_location_generator().segment_dir(),
+        least_visible_timestamp,
+        false,
+    )
+    .await?;
+    info!(
+        "list segments before gc_root for table {} takes {:?}, segment_dir: {:?}, least_visible_timestamp: {:?}, segments: {:?}",
+        fuse_table.get_table_info().desc,
+        start.elapsed(),
+        fuse_table.meta_location_generator().segment_dir(),
+        least_visible_timestamp,
+        slice_summary(&segments_before_gc_root)
+    );
+
+    let start = std::time::Instant::now();
+    let segments_to_gc: Vec<String> = segments_before_gc_root
+        .into_iter()
+        .filter(|s| !gc_root_segments.contains(s))
+        .collect();
+    info!(
+        "Filter segments to gc for table {} takes {:?}, segments_to_gc: {:?}",
+        fuse_table.get_table_info().desc,
+        start.elapsed(),
+        slice_summary(&segments_to_gc)
+    );
+
+    let start = std::time::Instant::now();
+    let segments_io =
+        SegmentsIO::create(ctx.clone(), fuse_table.get_operator(), fuse_table.schema());
+    let segments = segments_io
+        .read_segments::<SegmentInfo>(&gc_root.segments, false)
+        .await?;
+    let mut gc_root_blocks = HashSet::new();
+    for segment in segments {
+        gc_root_blocks.extend(segment?.blocks.iter().map(|b| b.location.0.clone()));
+    }
+    info!(
+        "read segments for table {} takes {:?}",
+        fuse_table.get_table_info().desc,
+        start.elapsed(),
+    );
+
+    let start = std::time::Instant::now();
+    let blocks_before_gc_root = list_until_timestamp(
+        fuse_table,
+        &fuse_table.meta_location_generator().block_dir(),
+        least_visible_timestamp,
+        false,
+    )
+    .await?;
+    info!(
+        "list blocks before gc_root for table {} takes {:?}, block_dir: {:?}, least_visible_timestamp: {:?}, blocks: {:?}",
+        fuse_table.get_table_info().desc,
+        start.elapsed(),
+        fuse_table.meta_location_generator().block_dir(),
+        least_visible_timestamp,
+        slice_summary(&blocks_before_gc_root)
+    );
+
+    let start = std::time::Instant::now();
+    let blocks_to_gc: Vec<String> = blocks_before_gc_root
+        .into_iter()
+        .filter(|b| !gc_root_blocks.contains(b))
+        .collect();
+    info!(
+        "Filter blocks to gc for table {} takes {:?}, blocks_to_gc: {:?}",
+        fuse_table.get_table_info().desc,
+        start.elapsed(),
+        slice_summary(&blocks_to_gc)
+    );
+
+    let start = std::time::Instant::now();
+    let files_to_gc = snapshots_to_gc
+        .iter()
+        .chain(segments_to_gc.iter())
+        .chain(blocks_to_gc.iter())
+        .map(|p| p.to_string())
+        .collect::<Vec<_>>();
```

Comment:
also need to delete the bloom filter index, inverted index and agg_indexes files

https://github.com/datafuselabs/databend/blob/d6eaadb875a2906f8d303c3fa7682fee1635a052/src/query/storages/fuse/src/operations/gc.rs#L472-L480